textarea/textarea_test.go

```diff
@@ -606,10 +608,370 @@ func TestView(t *testing.T) {
+			modelFunc: func(m Model) Model {
+				m.Placeholder = "123456789012345678"
+				m.MaxWidth = 20
+				m.ShowLineNumbers = false
+
+				return m
+			},
+			expected: heredoc.Doc(`
+				> 123456789012345678
+				> ~
+				> ~
+				> ~
+				> ~
+				> ~
+			`),
+		},
+		{
+			name: "placeholder single line that is longer than max width at limit plus one",
+			modelFunc: func(m Model) Model {
+				m.Placeholder = "1234567890123456789"
+				m.MaxWidth = 20
+				m.ShowLineNumbers = false
+
+				return m
+			},
+			expected: heredoc.Doc(`
+				> 123456789012345678
+				> 9
+				> ~
+				> ~
+				> ~
+				> ~
+			`),
+		},
+		{
+			name: "placeholder single line that is longer than max width with line numbers at limit",
+			modelFunc: func(m Model) Model {
+				m.Placeholder = "12345678901234"
+				m.MaxWidth = 20
+				m.ShowLineNumbers = true
+
+				return m
+			},
+			expected: heredoc.Doc(`
+				>   1 12345678901234
+				> ~
+				> ~
+				> ~
+				> ~
+				> ~
+			`),
+		},
+		{
+			name: "placeholder single line that is longer than max width with line numbers at limit plus one",
+			modelFunc: func(m Model) Model {
+				m.Placeholder = "123456789012345"
+				m.MaxWidth = 20
+				m.ShowLineNumbers = true
+
+				return m
+			},
+			expected: heredoc.Doc(`
+				>   1 12345678901234
+				>     5
+				> ~
+				> ~
+				> ~
+				> ~
+			`),
+		},
+		{
+			name: "placeholder multiple lines that are longer than max width at limit",
+			modelFunc: func(m Model) Model {
+				m.Placeholder = "123456789012345678\n123456789012345678"
+				m.MaxWidth = 20
+				m.ShowLineNumbers = false
+
+				return m
+			},
+			expected: heredoc.Doc(`
+				> 123456789012345678
+				> 123456789012345678
+				> ~
+				> ~
+				> ~
+				> ~
+			`),
+		},
+		{
+			name: "placeholder multiple lines that are longer than max width at limit plus one",
+			modelFunc: func(m Model) Model {
+				m.Placeholder = "1234567890123456789\n1234567890123456789"
+				m.MaxWidth = 20
+				m.ShowLineNumbers = false
+
+				return m
+			},
+			expected: heredoc.Doc(`
+				> 123456789012345678
+				> 9
+				> 123456789012345678
+				> 9
+				> ~
+				> ~
+			`),
+		},
+		{
+			name: "placeholder multiple lines that are longer than max width with line numbers at limit",
+			modelFunc: func(m Model) Model {
+				m.Placeholder = "12345678901234\n12345678901234"
+				m.MaxWidth = 20
+				m.ShowLineNumbers = true
+
+				return m
+			},
+			expected: heredoc.Doc(`
+				>   1 12345678901234
+				>     12345678901234
+				> ~
+				> ~
+				> ~
+				> ~
+			`),
+		},
+		{
+			name: "placeholder multiple lines that are longer than max width with line numbers at limit plus one",
+			modelFunc: func(m Model) Model {
+				m.Placeholder = "123456789012345\n123456789012345"
+				m.MaxWidth = 20
+				m.ShowLineNumbers = true
+
+				return m
+			},
+			expected: heredoc.Doc(`
+				>   1 12345678901234
+				>     5
+				>     12345678901234
+				>     5
```

Comment:
I can't describe how awesome this test is! Thank you so much!

Reply:
We need a lot more boundary checking in the tests.

Feel free to recommend any others you think you are useful (they are pretty quick to code).